src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -427,6 +428,8 @@ def find_or_create_tag(self, tag, root_name=None, **new_symbol_args):
         :param str tag: tag identifier.
         :param str root_name: optional name of the new symbol if it needs \
                               to be created. Otherwise it is ignored.
+        :param bool exact_name: whether to disallow renaming of the
+                                     created symbol.
```

Comment:
When moving it to new_symbol, I would prefer it to be "allow_renaming" with the docstring "whether the allow the generated symbol name to be different that the provided root_name. Defaults to True."

Tbh, it would also be useful to leave the docstring descriptions here together with the new_symbol "shadowing" and "symbol_type", otherwise it's difficult to know that they exist.